[DOCS-1956] Update docs for hashed API keys

mdlinville opened the pull request:
## Description
[DOCS-1956] Update docs for hashed API keys

- Update references to `authorize` to point to User Settings instead
- Create reusable snippets for finding, creating, and storing org and team service accounts and API keys
- Update content in the snippets to align with hashed API key updates
- Update pages to use the snippets
- Mention minimum SDK versions
- Consolidate articles about rotating and finding API keys add redirect
- Copyedits

## Testing
- [x] Test SDKs locally
- [x] Local build succeeds without errors (`mint dev`)
- [x] Local link check succeeds without errors (`mint broken-links`)
- [x] PR tests succeed



[DOCS-1956]: https://wandb.atlassian.net/browse/DOCS-1956?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mdlinville committed:
[DOCS-1956] Update docs for hashed API keys

- Create reusable snippets for finding,
  creating, and storing API keys
- Update content in the snippets to align
  with hashed API key updates
- Update pages to use the snippets
- Consolidate support articles about finding and rotating API keys

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1956

mdlinville committed:
Adjustments for UI changes and the settings landing page

jennwandb commented on snippets/en/_includes/api-key-create.mdx:
```diff
@@ -0,0 +1,13 @@
+
+import ApiKeyViewOnceWarning from "/snippets/en/_includes/api-key-view-once-warning.mdx";
+
+1. Navigate to [wandb.ai/authorize](https://wandb.ai/authorize) and log in if necessary.
```
Instead, let's encourage creating the API key from User Settings

jennwandb commented on snippets/en/_includes/api-key-create.mdx:
```diff
@@ -0,0 +1,13 @@
+
+1. Navigate to [wandb.ai/authorize](https://wandb.ai/authorize) and log in if necessary.
+2. Click **Create new API key**.
+3. Provide a descriptive name for your API key. This cannot be updated after the API key is created.
```
after receiving some feedback, we're working on adding the ability to edit API Key names asap! can we remove the sentence "This cannot be updated after the API key is created." ?

jennwandb commented on snippets/en/_includes/api-key-find.mdx:
```diff
@@ -0,0 +1,12 @@
+To find your API key:
+
+1. Navigate to [wandb.ai/authorize](https://wandb.ai/authorize) and log in if necessary.
```
For finding an API key, let's direct users to User Settings rather than /authorize

jennwandb commented on models/support/find_api_key.mdx:
```diff
@@ -1,12 +1,45 @@
+
+### Personal API keys
+
+1. Navigate to [wandb.ai/authorize](https://wandb.ai/authorize) or go to **User Settings** and scroll to the **API Keys** section.
```
A Personal API key can be deleted from User Settings. i think it's better to not mention /authorize here

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1956

mdlinville committed:
Jenn's feedback, other adjustments

- Change all references to  endpoint to User Settings
- Remove mention that description can't be updated
- Add more snippets for creating, finding team and org service
  accounts and API keys
- Adjust flow for getting to org API Keys tab to avoid placeholders in URLs
- Fix countable noun errors
- Fix URLs that are not marked up properly
- Fix minor Python notebook syntax errors

jennwandb commented on scripts/gpt-editor.py:
```diff
@@ -127,7 +127,7 @@ def generate_prompt(content, vale_output):
 - Commas and periods must go inside quotation marks.
 - Headings must use sentence-style capitalization.
-- Do NOT change URLs in the text (for example wandb.ai/authorize)
+- Do NOT change URLs in the text (for example wandb.ai/settings)
```
i'm not familiar with this doc, but the change seems fine?

jennwandb commented on weave/cookbooks/Intro_to_Weave_Hello_Eval.mdx:
```diff
@@ -34,9 +34,9 @@ import weave
 print("---")
 print(
-    "You can find your Weights and Biases API key here: https://wandb.ai/settings#api"
+    "Create a W&B API key at: https://wandb.ai/settings#api"
```
i recently found the working link is https://wandb.ai/settings#apikeys

jennwandb reviewed (approved):
this is awesome, thank you for all the updates!!

mdlinville committed:
Jenn's new feedback about #apikeys anchor, fix a couple unrelated Markdown formatting errors

dbrian57 commented on models/integrations/fastai.mdx:
```diff
@@ -1,19 +1,19 @@
+
+import ApiKeyCreateStreamlined from "/snippets/en/_includes/api-key-create-streamlined.mdx";
+
+If you're using **fastai** to train your models, W&B has an easy integration using the `WandbCallback`. Explore the details in [interactive docs with examples →](https://app.wandb.ai/borisd13/demo_config/reports/Visualize-track-compare-Fastai-models--Vmlldzo4MzAyNA)
```
```suggestion
You can integrate **fastai** with W&B using the `WandbCallback` class. Check out these [interactive docs with examples](https://app.wandb.ai/borisd13/demo_config/reports/Visualize-track-compare-Fastai-models--Vmlldzo4MzAyNA) for more details.
```

I'm wary of calling anything "easy"

dbrian57 commented on models/support/find_api_key.mdx:
```diff
@@ -1,12 +1,44 @@
+
+To revoke access, delete the API key:
+
+### Personal API keys
```
Should we put these in tabs instead?

mdlinville replied:
Good idea, done

dbrian57 commented on platform/app/settings-page.mdx:
```diff
@@ -5,6 +5,19 @@ title: Settings
 
-Within your individual user account you can edit: your profile picture, display name, geography location, biography information, emails associated to your account, and manage alerts for runs. You can also use the settings page to link your GitHub repository and delete your account. For more information, see [User settings](/platform/app/settings-page/user-settings/).
+Within your individual user account you can:
+- Edit your profile, including your profile picture, display name, geography location, biography information, and email addresses associated to your account.
```
```suggestion
- Edit your profile, including your profile picture, display name, geographic location, biography information, and email addresses associated with your account.
```

dbrian57 commented on snippets/en/_includes/api-key-find.mdx:
```diff
@@ -0,0 +1,11 @@
+
+1. Navigate to [User Settings](https://wandb.ai/settings) and log in if necessary.
+2. Alternatively, access your user settings:
+   - Click your user profile icon in the upper right corner.
```
```suggestion
   - Click your user profile icon in the upper-right corner.
```

mdlinville replied:
I'll take this suggestion here, but this is an area where we could do a docs-wide refactor.

dbrian57 commented on snippets/en/_includes/api-key-security.mdx:
```diff
@@ -0,0 +1,54 @@
+
+- **Secrets manager**: Use a dedicated secrets management system such as [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/), [HashiCorp Vault](https://www.vaultproject.io/), [Azure Key Vault](https://azure.microsoft.com/en-us/products/key-vault), or [Google Secret Manager](https://cloud.google.com/security/products/secret-manager).
+- **Password manager**: Use a reputable password manager application.
+- **OS-level keychains**: Store keys in macOS Keychain, Windows Credential Manager, or Linux secret service. Not typically appropriate for production.
```
```suggestion
- **OS-level keychains**: Store keys in macOS Keychain, Windows Credential Manager, or Linux secret service. Not suggested for production.
```

dbrian57 commented on snippets/en/_includes/api-key-view-once-warning.mdx:
```diff
@@ -0,0 +1,3 @@
+<Warning>
+The full API key is only shown once at creation time. After you close the dialog, you will not be able to view the full API key again. Only the key ID (first part of the key) will be visible in your settings. If you lose the full API key, you must create a new API key.
```
```suggestion
The full API key is only shown once at creation time. After you close the dialog, you cannot view the full API key again. Only the key ID (first part of the key) is visible in your settings. If you lose the full API key, you must create a new API key.
```
This is future tense that I think we can easily eliminate.

dbrian57 commented on weave/cookbooks/source/learn-weave-with-inference.ipynb:
```diff
@@ -32,49 +21,49 @@
-      "metadata": {
-        "id": "gfpToYNRdk1R"
-      }
+      ]
```
what's with all the other edits throughout this doc?

mdlinville replied:
I'm not sure why it happened, but it's just re-ordering sibling JSON keys in lexical order. I think there were some minor validity errors in the notebook that I needed to fix. I don't think there is any impact on the output from the re-ordering.

ngrayluna replied:
@mdlinville Can you clear the metadata? There are a couple of packages out there like nbdev, nb-clean, etc. that you can use to do this.

mdlinville replied:
Done in place, and will create a separate PR to commit a notebook cleaning script to `scripts/`. I won't add a CI check for now.

mdlinville replied:
Durable script available for review in #2020

dbrian57 commented on support-tags.mdx:
```diff
@@ -2431,14 +2427,9 @@ Among other things, service accounts are useful for tracking automated jobs logg
-* Select **Generate API key (Built-in)** as the authentication method
-* Press the **Create** button
-* Click the **Copy API key** button for the newly created service account and store it in a secret manager or another safe but accessible location
+The full API key is only shown once at creation time. After you close the dialog, you will not be able to view it again.
```
```suggestion
The full API key is only shown once at creation time. After you close the dialog, you cannot view it again.
```

mdlinville replied:
Done (Github didn't make this easy to apply for some reeason)

dbrian57 reviewed (approved):
Some minor notes and questions. Approving as to not block you.

ngrayluna commented on platform/app/settings-page.mdx:
```diff
@@ -5,6 +5,19 @@ title: Settings
 sidebarTitle: Overview
 ---
```
Add page links to bullet point items for easier navigation?

mdlinville committed:
Dan's feedback 1/2

Co-authored-by: Dan Brian <daniel.brian@gmail.com>

mdlinville committed:
Merge branch 'main' into DOCS-1956

mdlinville committed:
Dan's feedback 2/2

mdlinville committed:
Noah's feedback

mdlinville committed:
Noah's feedback: Clean metadata from the notebook

ngrayluna commented on weave/cookbooks/source/learn-weave-with-inference.ipynb:
```diff
@@ -1,372 +1,345 @@
-    "language_info": {
-      "name": "python"
-    }
+ "cells": [
```
Why is GitHub showing the diff like you are replacing the entire content?

mdlinville replied:
Well, removing the cell-level metadata changed a lot of lines, compared to the very small substantive content change to the notebook. Previously the metadata change was mixed into the content change, but I have now separated them. It will still look just about as noisy if you are reviewing the cumulative diff, but if you review the before that one, the diff will be more targeted for you to review.

mdlinville committed:
Clean notebook metadata with nb-clean

Remove execution counts, outputs, and cell metadata from the notebook
using the nb-clean script. This makes diffs cleaner and avoids noisy
metadata changes in version control.

The notebook retains the content change from the previous commit
(adding API key creation link).

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1956

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1956' into DOCS-1956

Resolved conflict in notebook by keeping our version cleaned with nb-clean.
The old commit used manual metadata cleanup, which we replaced with
the official nb-clean script for consistency.

mdlinville committed:
Merge branch 'main' into DOCS-1956

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1956

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1956' into DOCS-1956
